### PR TITLE
Update sample-app to use kubectl set image

### PR DESCRIPTION
**Pull request by mo-mughrabi**
The logic replaced, was going through yaml file and updating image name. 

This change, could update the image directly without extra code, unless it was put for another reason.

### CODE REVIEW

**Comment by viglesiasce** [comment]
Looks good @mo-mughrabi! Can you sign the CLA?

**Comment by mo-mughrabi** [comment]
Hi @viglesiasce, I just signed the CLA 👍

File: sample-app/Jenkinsfile [hunk 1]
```diff
@@ -19,20 +19,12 @@ node {
   switch (env.BRANCH_NAME) {
     // Roll out to staging
     case "staging":
-        // Change deployed image in staging to the one we just built
-        sh("sed -i.bak 's#gcr.io/cloud-solutions-images/gceme:1.0.0#${imageTag}#' ./k8s/staging/*.yaml")
-        sh("kubectl --namespace=production apply -f k8s/services/")
-        sh("kubectl --namespace=production apply -f k8s/staging/")
-        sh("echo http://`kubectl --namespace=production get service/${feSvcName} --output=json | jq -r '.status.loadBalancer.ingress[0].ip'` > ${feSvcName}")
+        sh("kubectl --namespace=staging set image deployments/${appName} ${appName}=${imageTag}"
```
Comment (viglesiasce) [inline sample-app/Jenkinsfile]: There is a syntax error here with a missing ), same below.

Comment (viglesiasce) [inline sample-app/Jenkinsfile]: The namespace should be production in both cases as they are deployed side by side in a canary configuration.

File: sample-app/Jenkinsfile [hunk 2]
```diff
@@ -19,20 +19,12 @@ node {
   switch (env.BRANCH_NAME) {
     // Roll out to staging
     case "staging":
-        // Change deployed image in staging to the one we just built
-        sh("sed -i.bak 's#gcr.io/cloud-solutions-images/gceme:1.0.0#${imageTag}#' ./k8s/staging/*.yaml")
-        sh("kubectl --namespace=production apply -f k8s/services/")
-        sh("kubectl --namespace=production apply -f k8s/staging/")
-        sh("echo http://`kubectl --namespace=production get service/${feSvcName} --output=json | jq -r '.status.loadBalancer.ingress[0].ip'` > ${feSvcName}")
```
Comment (viglesiasce) [inline sample-app/Jenkinsfile]: I'd prefer to leave these lines in.

File: sample-app/Jenkinsfile [hunk 3]
```diff
@@ -19,20 +19,12 @@ node {
   switch (env.BRANCH_NAME) {
     // Roll out to staging
     case "staging":
-        // Change deployed image in staging to the one we just built
-        sh("sed -i.bak 's#gcr.io/cloud-solutions-images/gceme:1.0.0#${imageTag}#' ./k8s/staging/*.yaml")
-        sh("kubectl --namespace=production apply -f k8s/services/")
```
Comment (viglesiasce) [inline sample-app/Jenkinsfile]: If these 2 lines are removed then changes to labels and deployment args will not be picked up.

File: sample-app/Jenkinsfile (hunk 1 above)
Comment (eggie5) [inline sample-app/Jenkinsfile]: @viglesiasce so are you saying this tutorial isn't following the staging/prod pattern anymore, but using a canary pattern? I was confused why the staging section of the Jenkins config was using the production namespace too...

Comment (viglesiasce) [inline sample-app/Jenkinsfile]: Correct, this tutorial takes a single service and deploys 2 matching deployments behind it. I opened #11 a while back and have been remiss in making that change. I will prioritize it.

**Comment by viglesiasce** [comment]
Closing.

**Pull request closed without merging**